core/Pos/Core/Context.hs

```diff
@@ -9,9 +24,57 @@ module Pos.Core.Context
 
 import           Universum
 
-import           Pos.Core.Address (addressHash, makePubKeyAddress)
-import           Pos.Core.Types   (Address, StakeholderId)
-import           Pos.Crypto       (PublicKey, SecretKey, toPublic)
+import           Control.Lens       (makeLenses)
+import           Data.Reflection    (Given (..), give)
+
+import           Pos.Core.Address   (addressHash, makePubKeyAddress)
+import           Pos.Core.Types     (Address, BlockCount (..), StakeholderId, SlotCount)
+import           Pos.Crypto         (PublicKey, SecretKey, toPublic)
+import           Pos.Core.Constants (staticBlkSecurityParam)
+
+-- | Core constants. They should be really constant and never change.
+data CoreConstants = CoreConstants
+    { _ccBlkSecurityParam :: !BlockCount
+    }
+
+makeLenses ''CoreConstants
+
+type HasCoreConstants = Given CoreConstants
+
+giveConsts :: BlockCount -> (HasCoreConstants => r) -> r
```

Comment:
You could look at the definition of `isBootstrapEra` and realize which argument is current epoch and which one is unlock epoch.
But you didn't do it and introduced a bug (https://github.com/input-output-hk/cardano-sl/pull/1298).
There are 100500 examples when someone confuses order of arguments.